server/data/mappers/actionPlanMapper.ts

```diff
@@ -1,19 +1,32 @@
-import type { ActionPlanResponse, GoalResponse, StepResponse } from 'educationAndWorkPlanApiClient'
+import type { ActionPlanResponse, GetGoalsResponse, GoalResponse, StepResponse } from 'educationAndWorkPlanApiClient'
 import type { ActionPlan, Goal, Step } from 'viewModels'
 import dateComparator from '../../routes/dateComparator'
 
 const toActionPlan = (actionPlanResponse: ActionPlanResponse, problemRetrievingData: boolean): ActionPlan => {
-  const goalReferencesInCreationDateOrder = goalReferencesSortedByCreationDate(actionPlanResponse.goals)
+  const goals = toOrderedGoals(actionPlanResponse.goals)
   return {
     prisonNumber: actionPlanResponse.prisonNumber,
-    goals: actionPlanResponse.goals.map((goal: GoalResponse) => {
```

Comment:
Oh my days! This was some nasty code! I think I wrote that! 😬 🤐 
It totally didn't do what I thought it did! 🤣 It was only seeing your change and how that works that made me realise that! 🤣 
Nice change; thank you 👍